### DISCUSSION TITLE
Is there a way for me to write middleware which can detect whether a user is making a Query or a Mutation?

### DISCUSSION THREAD

**Discussion by caldempsey** [99designs/gqlgen#1819] [Q&A]
### What happened?
Attempted to find a method call to determine wheteher a user is making a query or mutation in the standard library to append to my fancy new Datadog middleware I hand rolled.

### What did you expect?
A method call to determine whether a user is making a query or sending a mutation from OperationContext.

### Minimal graphql.schema and models to reproduce
One query, one mutation, inspect midddleware library

### versions
 - `gqlgen version`? latest 
 - `go version`? latest
 - dep or go modules? mod

**Comment by ngocdiep** [comment]
I think you can refer here:
https://github.com/99designs/gqlgen/blob/master/docs/content/recipes/authentication.md

**Comment by alexsn** [comment]
See this [reference](https://github.com/facebookincubator/ent-contrib/blob/master/entgql/transaction.go#L80-L82).

**Comment by caldempsey** [comment]
Thank you friends! I will check these out! Had to move to other user stories but will report ASAP.

**Comment by caldempsey** [comment]
This worked great
